src/Microsoft.AspNet.Mvc.Razor/RazorTextWriter.cs

```diff
@@ -51,6 +52,8 @@ public override Encoding Encoding
 
         private TextWriter TargetWriter { get; set; }
 
+        private IHtmlEncoder HtmlEncoder { get; set; } = new HtmlEncoder();
```

Comment:
They are pretty cheap, but this is wrong for a different reason - this is a service that we get from DI.